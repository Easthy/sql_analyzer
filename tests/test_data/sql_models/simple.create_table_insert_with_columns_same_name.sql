BEGIN TRANSACTION;

CREATE TABLE IF NOT EXISTS simple.create_insert_with_columns_same_name (
    user_id INTEGER,
    registration_date DATE ENCODE DELTA
)
    DISTSTYLE KEY
    DISTKEY(user_id)
    SORTKEY(registration_date)
;

INSERT INTO simple.create_insert_with_columns_same_name (
    user_id,
    registration_date
)
SELECT mysql_users.user_id,
       mysql_users.registration_date
  FROM mwl.mysql_users
;

END TRANSACTION;