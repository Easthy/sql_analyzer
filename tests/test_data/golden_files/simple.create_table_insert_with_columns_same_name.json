{
  "nodes": [
    { "id": "col:simple.create_insert_with_columns_same_name.user_id" },
    { "id": "col:simple.create_insert_with_columns_same_name.registration_date" },
    { "id": "tbl:simple.create_insert_with_columns_same_name" },
    { "id": "tbl:mwl.mysql_users" },
    { "id": "col:mwl.mysql_users.user_id" },
    { "id": "col:mwl.mysql_users.registration_date" }
  ],
  "links": [
    { "source": "tbl:simple.create_insert_with_columns_same_name", "target": "col:simple.create_insert_with_columns_same_name.user_id", "type": "contains_column" },
    { "source": "tbl:simple.create_insert_with_columns_same_name", "target": "col:simple.create_insert_with_columns_same_name.registration_date", "type": "contains_column" },
    { "source":"tbl:simple.create_insert_with_columns_same_name", "target":"tbl:mwl.mysql_users", "type":"table_dependency" },
    { "source": "tbl:mwl.mysql_users", "target": "col:mwl.mysql_users.user_id", "type": "contains_column" },
    { "source": "tbl:mwl.mysql_users", "target": "col:mwl.mysql_users.registration_date", "type": "contains_column" },
    { "source": "col:simple.create_insert_with_columns_same_name.user_id", "target": "col:mwl.mysql_users.user_id", "type": "column_dependency" },
    { "source": "col:simple.create_insert_with_columns_same_name.registration_date", "target": "col:mwl.mysql_users.registration_date", "type": "column_dependency" }
  ]
}